src/react.js

```diff
@@ -151,18 +151,27 @@ function createContext () {
     }
   }
 
-  function differ (mappedProps, nextMappedProps) {
+  function differ (mappedProps, nextMappedProps, { objects = false } = {}) {
```

Comment:
I don't get what the `objects` option does

Reply:
Yeah, I think this is actually wrong.

The whole reason I had to update this `differ` helper is to account for diffing primitives and not only 2 props objects or 2 state objects, which are always objects `{}` that can be compared shallowly. But now with hooks, if you call `useAtom(state => state.count)` and count is a  number, you need to diff it differently.

Basically, I think, objects need to be diffed shallowly and everything else by reference. But I can't just pass this option in like I did here, because if you map `useAtom(state => state.user)` where user is an object, this won't work as intended.

I'll work on a test and a fix.

Reply:
Fixed now, if these are plain objects, I compare shallowly, otherwise, we compare with ===.